src/ansys/fluent/core/quantity/quantity.py

```diff
@@ -0,0 +1,332 @@
+import ansys.fluent.core.quantity as q
+
+
+class Quantity(float):
+    """Quantity instantiates physical quantities using their real values and
+    units. All the instances of this class are converted to base SI units system to have
+    consistency in arithmetic operations.
+
+    Parameters
+    ----------
+    value : int | float
+        Real value of quantity.
+    units : str
+        Unit string representation of quantity.
+    quantity_map : dict
+        Quantity map representation of quantity.
+    dimensions : list
+        Dimensions representation of quantity.
+
+    Methods
+    -------
+    to()
+        Converts to given unit string.
+    convert()
+        Converts to given unit system.
+
+    Returns
+    -------
+    Quantity instance.
+    """
+
+    def __new__(cls, value, units=None, quantity_map=None, dimensions=None):
+        if (
+            (units and quantity_map)
+            or (units and dimensions)
+            or (quantity_map and dimensions)
+        ):
+            raise ValueError(
+                "Quantity only accepts 1 of the following: units, quantity_map, dimensions"
+            )
+
+        _units_table = q.UnitsTable()
+        _value = float(value)
+
+        if units or units == "":
```

Comment:
There is a subtle logical discrepancy between this assignment code and the above validation code. In both places, we should apply the simple condition `units is not None`.

Reply:
Originally was having issues with evaluating the truthy value on empty vs non-empty strings. Fixed now checking `units is not None`.

https://github.com/ansys/pyfluent/blob/911b052f1e7162a45f2bc72965029fcf5045470b/src/ansys/fluent/core/quantity/quantity.py#L45-L46

https://github.com/ansys/pyfluent/blob/911b052f1e7162a45f2bc72965029fcf5045470b/src/ansys/fluent/core/quantity/quantity.py#L79-L81